### ISSUE TITLE
Adafruit CircuitPython Library Download Stats failing

### ISSUE THREAD

**Issue by kattni** [adafruit/adabot#162]
Something is causing the PyPI download stats to fail.

The list currently begins like this: 
```
Adafruit_CircuitPython_Bundle downloads for the past week:
 adafruit-circuitpython-bundle-5.x-mpy: 627
 adafruit-circuitpython-bundle-4.x-mpy: 149
 adafruit-circuitpython-bundle-py: 133
 adafruit-circuitpython-bundle-examples: 67

Adafruit CircuitPython Library PyPi downloads:

| Library                                                | Last Week   | Total |   
|:-------                                                |:--------:   |:-----:|   
| Adafruit_CircuitPython_FancyLED                        | 279         | 0 |       
| Adafruit_CircuitPython_PCA9685                         | 143         | 0 |       
| Adafruit_CircuitPython_Motor                           | 119         | 0 |       
| Adafruit_CircuitPython_PN532                           | 83          | 0 |       
| Adafruit_CircuitPython_framebuf                        | 78          | 0 |       
| Adafruit_CircuitPython_BME280                          | 69          | 0 |       
| Adafruit_CircuitPython_binascii                        | 47          | 0 |       
| Adafruit_CircuitPython_BLE                             | 43          | 0 |       
| Adafruit_CircuitPython_PCT2075                         | 37          | 0 |       
| Adafruit_CircuitPython_Touchscreen                     | 36          | 496 |     
| Adafruit_CircuitPython_hashlib                         | 34          | 211 |     
| Adafruit_CircuitPython_MLX90640                        | 34          | 0 |       
| Adafruit_CircuitPython_BNO055                          | 31          | 0 |       
| Adafruit_CircuitPython_MAX31856                        | 29          | 543 |     
[...]
```
And continues halfway down as follows:
``` * Failed to retrieve stats for the following libraries:
   * Adafruit_CircuitPython_MLX90614
   * Adafruit_CircuitPython_SharpMemoryDisplay
   * Adafruit_CircuitPython_MCP9600
   * Adafruit_CircuitPython_STMPE610
   * Adafruit_CircuitPython_MAX31855
   * Adafruit_CircuitPython_BLE_Cycling_Speed_and_Cadence
   * Adafruit_CircuitPython_DS2413
   * Adafruit_CircuitPython_Trellis
   * Adafruit_CircuitPython_Thermistor
[...]
```

**Comment by sommersoft** [comment]
`pypistats.org` has instituted rate-limiting: https://pypistats.org/api/#rate-limiting.

There is no published subscription to increase the rate-limit, that I can find.

**Comment by kattni** [comment]
@sommersoft Hmm. Any suggestions for other options? If not, we should consider disabling what we have as it's clearly not going to function moving forward.

**Comment by sommersoft** [comment]
Only options I can think of:
1. Reaching out to the `pypistats.org` maintainer(s) with some support offering. @tannewt had suggested this a while back when the service went down for a while.

2. Roll our own with the Google BigQuery. How hard could it be? (Famous last words...)

PiWheels is getting _closer_ to their API as well, last I looked. Which will have better stats for our target audience (SBCs).

**Comment by kattni** [comment]
I was wondering about option 2 after looking at the `pypistats.org` site. But... famous last words indeed.

**Comment by sommersoft** [comment]
So, I dug into using BigQuery a little more. Using the web interface, I've got a query formulated that appears to be fairly accurate:
```sql
SELECT
  COUNT(*) AS num_downloads,
  file.project AS `lib`
FROM `the-psf.pypi.downloads*`
WHERE
  file.project LIKE 'adafruit-%'
  AND _TABLE_SUFFIX
    BETWEEN FORMAT_DATE(
      '%Y%m%d', DATE_SUB(CURRENT_DATE(), INTERVAL 7 DAY))
    AND FORMAT_DATE('%Y%m%d', CURRENT_DATE())
GROUP BY `lib`
ORDER BY `num_downloads` DESC
```
Example (278 rows returned in total):

Row | num_downloads | lib
-- | -- | --
1 | 9435 | adafruit-pureio
2 | 5852 | adafruit-blinka
3 | 4575 | adafruit-platformdetect
4 | 1810 | adafruit-nrfutil
5 | 1558 | adafruit-gpio
6 | 1434 | adafruit-bbio
7 | 1412 | adafruit-circuitpython-busdevice
8 | 1318 | adafruit-circuitpython-esp32spi
9 | 1085 | adafruit-ampy
10 | 699 | adafruit-dht
11 | 696 | adafruit-circuitpython-ble


This only uses ~13.7GBs per query, which would allow for approx 74 free queries per month (1TB/month is free). Using `adafruit-%` in the `WHERE` clause includes non-circuitpython results. This was necessary to get Blinka and related info. If I set another clause with an `OR file.project = 'adafruit-blinka'`, the data-per-query shot up to over 900GB/query.

With that knowledge, I think we may be able to use the BigQuery directly in adabot. There is a Python library available, `google-cloud-bigquery`. Just need to solidify the authentication process, plug in the query string, and work out handling the result.
